proposed/2021/CentralPackageManagementTransitive.md

```diff
@@ -0,0 +1,201 @@
++ <PackageReference Include="PackageC" />
+</ItemGroup>
+```
+
+And add a corresponding `PackageVersion` item to Directory.Packages.props:
+```diff
+<ItemGroup>
+  <PackageVersion Include="PackageA" Version="1.0.0" />
+  <PackageVersion Include="PackageB" Version="2.0.0" />
++ <PackageVersion Include="PackageC" Version="5.0.0" />
+</ItemGroup>
+```
+
+The result is the same as before where the resolved `PackageC` is version **`5.0.0`** and affects the restored packages, reference assemblies passed to the compiler, files copied to the output directory, and dependencies in used during package creation.  The transitive dependency is no longer considered transitive and instead is an explicit dependency.
+
+### Proposal to allow transitive version selection based on specified PackageVersion items
+In the previous example, a user must specify an explicit `PackageReference` to a package to override the transitive version.  This elevates the reference from transitive to explicit and behaves as a user expects.  However, some users want a `PackageVersion` to automatically override a transitive version even without an explicit top-level `PackageReference`.  The reason for this feature request is for better **scale** and **performance**.
+
+#### Addressing Scale
+For larger repositories, requiring users to specify explicit `PackageReference` items in each project can become difficult.  This is because you may need to add it to dozens of projects in order to unify the version used across the repo.  Since CPVM is used to centralize package versions, it makes sense to have transitive reference version overrides to be implied when specifying a `PackageVersion`.  There are however at least two drawbacks of this approach:
+
+1. **Confusion** about "pinning" a transitive version - Many users consider overriding a transitive version to be pinning that version in the graph.  In some cases they consider this gesture only part of Restore and Build.  But since they've overridden a transitive version, this also flows to Pack which affects the versions used as dependencies.  When transitive versions are are "pinned", they are really just elevated to top-level dependencies which causes this confusion.  We will need to educate users on how this feature works so they understand that it is not really just pinning a version but instead is actually making it an implicit reference.
+
+3. How to opt out - If transitive version overrides are automatic when a `PackageVersion` is specified, there will need to be a way for a user to opt out of this behavior for one or more `PackageReference` items.  In previous versions of NuGet, a user could only opt into transitive version overrides by _adding_ a `PackageReference` item.  But to opt out of newer behavior, they will not have a `PackageReference` item to remove.  An MSBuild property could be used to opt an entire project out of transitive version overrides but that would not address a single `PackageReference` item.
```

Comment:
I don't think that a project-wide MSBuild property is needed...  And it can always be added later, if there is demand.  I believe that the overwhelming majority of users will want to keep transitive version pinning, with some rare exceptions for some individual packages.